zebra/zebra_nhg.c

```diff
@@ -1074,8 +1096,8 @@ static struct nhg_hash_entry *depends_find(const struct nexthop *nh, afi_t afi)
 static void depends_add(struct nhg_connected_tree_head *head,
 			struct nhg_hash_entry *depend)
 {
-	nhg_connected_tree_add_nhe(head, depend);
-	zebra_nhg_increment_ref(depend);
+	if (nhg_connected_tree_add_nhe(head, depend) == NULL)
```

Comment:
so as I'm reading this, it's a bit confusing: change the signature of the function so that as a side-effect the caller learns something. couldn't `nhg_connected_tree_add_nhe()` do this internally? that is, couldn't it either a) add a new object, or b) add a ref to an existing object?

Reply:
I don't think that `nhg_connected_tree_add_nhe()` should manage refcounting. It should add to a tree and be done there. Refcounting belongs to whatever calls that and then handles the success/failure. Hence, this function.

Reply:
Could we at least have a comment here then, about what's happening? something along the lines of "a NULL return means that 'depend' is already present, so we'll add a ref here"

Reply:
Could we at least have a comment here then, about what's happening? something along the lines of "a NULL return means that 'depend' is already present, so we'll add a ref here"

Reply:
Could we at least have a comment here then, about what's happening? something along the lines of "a NULL return means that 'depend' is already present, so we'll add a ref here"

Reply:
done